### PR TITLE
Add accordion

### PR DESCRIPTION
**Pull request by mfrachet**
# Accordion

## Description

An  accordion component

## Demo

Example on : [deployed story link]

## API

The most simple:

```jsx
<Accordion expanded={expanded} toggle={() => setExpanded((s) => !s)} id="acc-1">
  <AccordionToggle
    title="User informations"
    description="The following contains information about the current user"
  />
  <AccordionContent>
    <Box padding={3}>My name is John Duff</Box>
  </AccordionContent>
</Accordion>
```

The most complex:

```jsx
<Accordion expanded={expanded} toggle={() => setExpanded((s) => !s)} id="acc-4">
  <AccordionToggle
    variant="secondary"
    togglePosition="left"
    title="User informations 4"
    description="The following contains information about the current user 3"
    action={
      <IconButton onClick={() => console.log('edit')} title="Edit">
        <EditIcon />
      </IconButton>
    }
  />
  <AccordionContent>
    <Box padding={3}>My name is John Duff</Box>
  </AccordionContent>
</Accordion>
```

## Voiceover

Voiceover opening and closing the accordion component to demonstrate the labelling and description of the element

### CODE REVIEW

File: packages/strapi-design-system/src/Accordion/AccordionToggle.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+const ToggleButton = styled.button`
+  border: none;
+  background: transparent;
+  display: block;
+  width: 100%;
+  text-align: unset;
+  padding: 0;
+`;
+
+export const AccordionToggle = ({ title, description, as, variant, togglePosition, action, ...props }) => {
```
Comment (HichamELBSI) [inline packages/strapi-design-system/src/Accordion/AccordionToggle.js:20]: What is the utility of the `as` prop? Will it be used to update the heading level only?

Comment (HichamELBSI) [reply packages/strapi-design-system/src/Accordion/AccordionToggle.js:20]: If so, WDYT about renaming that prop to something like `headingLevel={2}`? And create a mapping `1 => h1`, `2 -> h2` to reduce the scope of that prop... 🤔

File: packages/strapi-design-system/src/Accordion/AccordionToggle.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+          {dropdownIcon}
+        </Row>
+      </ToggleButton>
+    </Box>
+  );
+};
+
+AccordionToggle.defaultProps = {
+  action: undefined,
+  as: 'span',
```
Comment (HichamELBSI) [inline packages/strapi-design-system/src/Accordion/AccordionToggle.js:101]: I just saw that 😅 , my comment about the headingLvl is probably irrelevant

**Review by HichamELBSI — CHANGES_REQUESTED** [review]
Very nice job!

File: packages/strapi-design-system/src/Accordion/AccordionToggle.js (hunk 1 above)
Comment (mfrachet) [reply packages/strapi-design-system/src/Accordion/AccordionToggle.js:20]: We can totally 😊 But what if it's not a heading?

**Pull request merged**